docs/configuration.md

```diff
@@ -0,0 +1,297 @@
+                vssKey: "WCEChSQx6z4OxYrHNYbu5GGztX4FBBxGMfzmO6C+xNTrDVw="
+        nonAvvmBalances: {}
+```
+
+Such type of genesis spec was used once to construct mainnet genesis
+data. It's not clear whether it will be ever used again.
+
+### System start time
+
+System start time is taken from configuration or command line
+option. It's taken from configuration if genesis is provided as
+genesis data (which always contains system start time) or if genesis
+is provided as genesis spec with `mainnetInitializer`. The reason why
+system start is part of `mainnetInitializer`, but not part of
+`testnetInitializer` is that mainnet is launched rarely (actually only
+once, but there is also staging), but other
+clusters are launched more often, and it's easier to change this value
+from CLI.
+
+### Tools
+
+There are some tools relevant to genesis data.
+
+* `cardano-keygen` has a command to dump all secret keys and similar
+  data to files. Usage: `cardano-keygen --system-start 0
+  --configuration-file <file> --configuration-key <key>
+  generate-keys-by-spec --genesis-out-dir <dir>`. This command will
+  generate dump secrets to `<dir>`. To deploy a cluster you need keys
+  of core nodes. In case of devnet, just use `generate-keys-by-spec`
+  to obtain these keys. They can be found in
+  `keys-testnet/rich`. Workflow for testnet is described
+  below. Workflow for mainnet (not sure if someone will ever need it)
+  is described in `scripts/prepare-genesis` (**TODO** move it here
+  maybe?).
+* `cardano-node-simple` and `cardano-node` have command line option to
+  dump genesis data (in JSON format). The option is
+  `--dump-genesis-data-to genesis.json` (data will be dumped to
+  `genesis.json`). It can be used to verify generated balances, for
+  example.
+
+### Generating genesis for testnet
+
+There is `testnet_public_full` configuration which is almost suitable
+for testnet. The only values to be changed are inside
+`testnetCustomStakeDistr`: `bootStakeholders` and `vssCerts`. They
+depend on secret keys of core nodes which shouldn't be publicly
+known. You need to generate secret keys and VSS certificates and put
+them into configuration. There is also `testnet_staging_full` which
+has different `k`, `protocolMagic` and some other values.
+
+To generate a secret key use `cardano-keygen --system-start 0
+generate-key --path <SECRET_KEY_PATH>`. Generate as many keys as there
+should be core nodes. Last line looks like this:
+
+> [keygen:INFO] Successfully generated primary key and dumped to secrets/testnet/node4.sk, stakeholder id: 39f2bb9fd75ac348e6c92467e61eb3e1418a394f5a2105f9e014b666, PK (base64): /fbQqCqUPdPImYA2djkGYMpj9HZGkDeKhTc/mLcPG7sdhJSR6Ou0sB3J5OII2VdWTXSHRM1cPdNwZsaNba4rcg==
+
+You should use stakeholder ids as keys in `bootStakeholders`
+map. Values are stakeholders' weights, having all weights equal to 1
+is fine.
+
+The second step is VSS certificates generation. Use `cardano-keygen
```

Comment:
What are VSS certificates?  Again, a pointer to background documentation would be helpful here.

Reply:
Added `Prerequisites` chapter in the beginning with the relevant link.